src/sentry/middleware/integrations/parsers/slack.py

```diff
@@ -82,19 +92,101 @@ class SlackRequestParser(BaseRequestParser):
+        # https://api.slack.com/reference/interaction-payloads/block-actions
+        action_ts = payload.get("actions")[0].get("action_ts")
+
+        integration = self.get_integration_from_request()
+        if not integration:
+            raise ValueError("integration not found")
+
+        slack_client = SlackSdkClient(integration_id=integration.id)
+        loading_modal = self.build_loading_modal(action_ts, title)
+
+        try:
+            slack_client.views_open(
+                trigger_id=payload["trigger_id"],
+                view=loading_modal,
+            )
+            metrics.incr(
+                SLACK_MIDDLE_PARSERS_SUCCESS_DATADOG_METRIC,
+                sample_rate=1.0,
+                tags={"type": action},
+            )
+        except SlackApiError:
+            metrics.incr(
+                SLACK_MIDDLE_PARSERS_FAILURE_DATADOG_METRIC,
+                sample_rate=1.0,
+                tags={"type": action},
+            )
+            logger_params = {
+                "integration_id": integration.id,
+                "action": action,
+            }
+            logger.exception("slack.control.view.open.failure", extra=logger_params)
+
     def get_async_region_response(self, regions: Sequence[Region]) -> HttpResponseBase:
         if self.response_url is None:
             return self.get_response_from_control_silo()
 
+        CONTROL_RESPONSE_ACTIONS = {
+            "resolve_dialog": lambda request, action: self.handle_dialog(
+                request, action, "Resolve Issue"
+            ),
+            "archive_dialog": lambda request, action: self.handle_dialog(
+                request, action, "Archive Issue"
+            ),
+            # Add more actions here, ie for buttons in modal
+        }
+
+        integration = self.get_integration_from_request()
+
+        # if we are able to  send a response to Slack from control itself to beat the 3 second timeout, we should do so
+        try:
+            if (
+                options.get("send-slack-response-from-control-silo")
+                and self.action_option in CONTROL_RESPONSE_ACTIONS
+            ):
+                CONTROL_RESPONSE_ACTIONS[self.action_option](self.request, self.action_option)
+        except ValueError:
+            logger.exception(
+                "slack.control.response.error",
+                extra={
+                    "integration_id": integration and integration.id,
+                    "action": self.action_option,
+                },
+            )
+
         convert_to_async_slack_response.apply_async(
             kwargs={
                 "region_names": [r.name for r in regions],
                 "payload": create_async_request_payload(self.request),
                 "response_url": self.response_url,
             }
         )
-        # We may want to enrich this with a waiting message
-        return HttpResponse(status=status.HTTP_202_ACCEPTED)
+
+        return HttpResponse(status=status.HTTP_200_OK)
```

Comment:
could we leave a note here as to why we respond with 200 OK even though we already rendered the modal?